client/src/sections/Fridge/Header/Navbar/Navbar.tsx

```diff
@@ -3,19 +3,33 @@ import {useHistory} from 'react-router-dom';
 import {Button, ButtonGroup} from '@shopify/polaris';
 
 import './Navbar.css';
+import {useDispatch} from "react-redux";
+import {logout} from "../../../../store/users/action";
+import {initialState} from "../../../../store/users/reducers";
 
 export default function Navbar() {
   const history = useHistory();
+  const dispatch = useDispatch();
 
+  const redirection = () => {
+    // console.log("Inside Redirection");
+    if(!initialState.isLoggedIn) {
+      alert("You Are Not Logged In!!!");
+      history.push('./login');
+    }
+    return null;
+  };
   const handleHomeClick = () => {
     history.push('./home');
   };
 
   const handleLogoutClick = () => {
     history.push('./login');
+    dispatch(logout());
   };
-
+  window.onload = redirection;
```

Comment:
By checking if the user is logged in the `<Fridge />` component you don't need this direct access to the window DOM element.